ambry-admin/src/main/java/com.github.ambry.admin/AdminBlobStorageService.java

```diff
@@ -356,9 +344,19 @@ public void onCompletion(String result, Exception exception) {
           logger.trace("Forwarding {} of {} to the router", restMethod, result);
           switch (restMethod) {
             case GET:
-              headForGetCallback.setBlobId(result);
-              headForGetCallback.markStartTime();
-              router.getBlobInfo(result, headForGetCallback);
+              RestUtils.SubResource subresource = RestUtils.getBlobSubResource(restRequest);
+              if (subresource == null || subresource.equals(RestUtils.SubResource.BlobInfo) || subresource
+                  .equals(RestUtils.SubResource.UserMetadata)) {
+                headForGetCallback.setBlobId(result);
+                headForGetCallback.markStartTime();
+                router.getBlobInfo(result, headForGetCallback);
+              } else {
+                switch (subresource) {
```

Comment:
Don't think it is required because the question again is "what is to be done in the default case"?